src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -1211,18 +1169,24 @@ private void processFailoverFailure(String message) throws SQLException {
   }
 
   private void setClusterId(String host, int port) {
+    String clusterId = null;
     if (!StringUtils.isNullOrEmpty(this.clusterIdSetting)) {
-      this.topologyService.setClusterId(this.clusterIdSetting);
+      clusterId = this.clusterIdSetting;
     } else if (this.isRdsProxy) {
       // Each proxy is associated with a single cluster so it's safe to use RDS Proxy Url as cluster identification
-      this.topologyService.setClusterId(host + ":" + port);
+      clusterId = host + ":" + port;
     } else if (this.isRds) {
       // If it's a cluster endpoint, or a reader cluster endpoint, then let's use it as the cluster ID
       String clusterRdsHostUrl = getRdsClusterHostUrl(host);
       if (!StringUtils.isNullOrEmpty(clusterRdsHostUrl)) {
-        this.topologyService.setClusterId(clusterRdsHostUrl + ":" + port);
+        clusterId = clusterRdsHostUrl + ":" + port;
       }
     }
+
+    if (clusterId != null) {
+      this.topologyService.setClusterId(clusterId);
+    }
+    metricsContainer.setClusterId(this.topologyService.getClusterId());
```

Comment:
Since you have this line (that is great!), you probably don't need any other changes in this method.